docs/eks-v3-migration.md

```diff
@@ -0,0 +1,150 @@
+  - **Improved Cost Efficiency**: `t3.medium` is marginally cheaper than `t2.medium` ($0.0416 vs $0.0464 per hour), providing better performance at a lower cost.
+
+
+- **New `operatingSystem` Property for Node Groups**  
+  A new `operatingSystem` property has been introduced for node groups, giving users the flexibility to choose which OS their nodes should run. Supported options include Amazon Linux 2023 (AL2023) and Bottlerocket, allowing you to select the OS that best fits your workloads and compliance needs.  
+    
+  - **`RECOMMENDED` Default Setting**: By default, the `RECOMMENDED` option will be used, which ensures your nodes are always running the OS recommended by AWS for optimal security and performance. This default setting will be kept in sync with AWS’s updates.
+
+
+- **EKS Cluster Addon Management: `vpc-cni`, `coredns` and `kube-proxy`**
+  The EKS cluster components `vpc-cni`, `coredns` and `kube-proxy` are now configured as EKS managed addons. This change eliminates the dependency on `kubectl` for managing these components and automates their updates, ensuring that clusters stay up to date without manual intervention.
+  This also simplifies management of clusters with private API endpoints because those components are managed via AWS APIs.
+  If you want to manage those EKS components yourself, you can do so by disabling them in the cluster component.
+    
+- **Support for Cluster Autoscaler and External Scaling**  
+  Pulumi EKS v3 introduces support for integrating with the Kubernetes cluster-autoscaler, which allows your cluster to scale dynamically based on resource needs. A new `ignoreScalingChanges` parameter has been added for node groups. This parameter allows Pulumi to ignore external scaling changes (e.g., changes to the desired size made by the cluster-autoscaler).
+    
+- **Cluster Version Tracking in `ManagedNodeGroup`**  
+  `ManagedNodeGroup` now automatically tracks the EKS cluster version, ensuring smoother version upgrades and preventing version mismatches. Since EKS only allows one minor version skew between the cluster control plane and the nodes, this automatic tracking helps prevent upgrade failures due to version incompatibility.
+
+# Migration
+
+## VPC CNI Component changes
+
+The VPC CNI cluster component is now configured as an EKS addon as mentioned in the “New Features” section above. This brings the following changes:
+
+- Removed `enableIpv6` input property. The component automatically configures the IP version now depending on whether the cluster is running in IPv4 or IPv6 mode.  
+- Removed `image`, `initImage`, `nodeAgentImage` input properties. The component now automatically selects an image registry in the cluster’s region to pull the images from.
+
+## Node Group Updates
```

Comment:
Should we mention this change somewhere

```
-      ├─ aws:cloudformation:Stack     cluster-nodes                    delete
```

Reply:
The effect this will have is described in the `NodeGroup` component deprecation section. (i.e. `Updates to the default node group will be done by first creating the new replacement nodes and then shutting down the old ones which will move pods to the new nodes.`)